codalab/lib/download_manager.py

```diff
@@ -43,11 +43,15 @@ def __init__(self, bundle_model, worker_model, bundle_store):
     def get_target_info(self, uuid, path, depth):
         """
         Returns information about an individual target inside the bundle, or
-        None if the target doesn't exist.
+        None if the target or bundle doesn't exist.
 
         For information about the format of the return value, see
         worker.download_util.get_target_info.
         """
+        # Return if invalid bundle reference
+        if uuid not in self._bundle_model.get_bundle_states([uuid]):
+            return
+
         if self._bundle_model.get_bundle_state(uuid) != State.RUNNING:
```

Comment:
I made appropriate changes such that I had to make the call only once. Other alternatives switched control flow to `else` block instead of returning. Please have a look at new code :smile: